### PR TITLE
executor: improve codes of stmt hints

### PR DESCRIPTION
**Pull request by foreyes**
### What problem does this PR solve? 

Improve codes of https://github.com/pingcap/tidb/pull/11809

### What is changed and how it works?

Reduce function cost when there is no sql hint.

Record last hint instead of save the whole list.

### Check List 

Tests 

 - Unit test

Code changes

 - None

Side effects

 - None

Related changes

 - None

Release note

 - Write release note for bug-fix or new feature.

### CODE REVIEW

**Comment by zz-jason** [comment]
should we cherry pick this pr to release-3.1?

**Comment by foreyes** [comment]
> should we cherry pick this pr to release-3.1?

Nope, seems we don't need these hints in 3.1.

File: executor/executor.go [hunk 1]
```diff
@@ -1385,61 +1385,65 @@ func extractStmtHintsFromStmtNode(stmtNode ast.StmtNode) []*ast.TableOptimizerHi
 }
 
 func handleStmtHints(hints []*ast.TableOptimizerHint) (stmtHints stmtctx.StmtHints, warns []error) {
-	var memoryQuotaHintList, noIndexMergeHintList, useToJAHintList, readReplicaHintList []*ast.TableOptimizerHint
+	if hints == nil {
```
Comment (lonng) [inline executor/executor.go]:
```suggestion
	if hints == nil || len(hints) < 1 {
```

Comment (lamxTyler) [reply executor/executor.go]: Then `len(hints) == 0` is enough.

**Comment by foreyes** [comment]
Could you approve again? @lamxTyler

**Comment by zz-jason** [comment]
> > should we cherry pick this pr to release-3.1?
> 
> Nope, seems we don't need these hints in 3.1.

Does release 3.1 have this performance issue?

**Pull request merged**